piptools/scripts/compile.py

```diff
@@ -214,11 +220,11 @@ def cli(
     annotate: bool,
     upgrade: bool,
     upgrade_packages: Tuple[str],
-    output_file: Optional[LazyFile],
+    output_file: Optional[click.Path],
```

Comment:
**Nitpick (feel free to ignore):** I would rename variable `output_file` (not option name) to `output_file_path`:

```suggestion
    output_file_path: Optional[click.Path],
```

```python
@click.option("-o", "--output-file", "output_file_path", ...)
```

since `output_file` is not a file object anymore. As a consequence, it would make diff less noisy.

Reply:
I'd initially gone that way, but ran into between 1 and 127 instances of `cli()` being invoked with a no longer valid keyword argument `output_file`, and decided to keep things simple, as I don't always catch `click` and `pytest` magic. You're welcome to make it work. Note also that `src_files` are `str`s.

Reply:
Alrighty, thanks for the explanation 👍🏼